packages/babel-plugin-transform-es2015-duplicate-keys/src/index.js

```diff
@@ -0,0 +1,38 @@
+import * as t from "babel-types";
+
+function getName(key) {
+  if (t.isIdentifier(key)) {
+    return key.name;
+  }
+  return key.value.toString();
+}
+
+export default function() {
+  return {
+    visitor: {
+      ObjectExpression(path) {
+        const { node } = path;
+        const plainProps = node
+          .properties
+          .filter(prop => !t.isSpreadProperty(prop) && !prop.computed);
+
+        const alreadySeenNames = {};
```

Comment:
We can use `Object.create(null);` here instead of `{}` and remove the `Object.prototype.hasOwnProperty` check. This is consistent with how we do hash maps elsewhere in the codebase.